streaming/base/dataset.py

```diff
@@ -306,6 +313,22 @@ def __init__(self,
             }
             for stream in streams:
                 stream.apply_default(default)
+        elif epoch_size_value:
+            # if there are no streams provided but the epoch_size is speficied
```

Comment:
nit: `speficied` -> `specified`. Also check the next line.

Reply:
rare typo when made twice in a row 😂

Reply:
modifying the `elif` case on this as discussed earlier